app/src/main/java/unipd/se18/ocrcamera/MignifyResolver.java

```diff
@@ -0,0 +1,111 @@
+    private static final String URL =
+            "https://mignify.p.mashape.com/gtins/v1.0/productsToGtin?gtin=";
+    //The authentication string you get from
+    //https://market.mashape.com/mignify/product-intelligence#
+    //after you subscribe to the Mignify API
+    private static final String MASHAPE_AUTH = "Nx432kOz2Wmsh83IAjUv6cUaxT7Rp1tDUbKjsnu7ueSdRFdOPK";
+    //Error text when connecting to the Mashape API
+    private static final String HTTP_ERROR = "Unable to connect to the server.";
+    //Error text if there's no product with entered EAN code in the database
+    private static final String NO_PRODUCT_ERROR =
+            "No product found in the database for that barcode.";
+    //Error text if there was an error parsing the JSON response
+    private static final String JSON_ERROR = "Error parsing the response from the server.";
+
+    /**
+     *
+     * @param EANCode string extracted from an EAN13 barcode
+     * @return the product relative to the input string
+     */
+    @Override
+    public String decodeEAN(String EANCode) {
+
+        //Adding to the URL request the EAN code taken from the barcode
+        String fullURL = URL + EANCode;
+        //HTTP response from get method to Mignify API
+        HttpResponse<JsonNode> response = null;
+        //Unirest library useful for making an HTTP request for the Mashape API
+        //More at http://unirest.io/java
+        try {
+            response = Unirest
+                    .get(fullURL)
+                    //Authentication header
+                    .header("X-Mashape-Key",
+                            MASHAPE_AUTH)
+                    //Accept and Content-Type selection header
+                    .header("Accept", "application/json")
+                    //Type of response we expect
+                    .asJson();
+        } catch (UnirestException e) {
+            Log.e("Unirest error", e.getMessage());
+        }
+        if (response == null) {
+            product = HTTP_ERROR;
+        } else {
+            try {
+                //Retrieve the parsed JSONObject from the Mashape response
+                JSONObject jsonResponse = response.getBody().getObject();
+                //Getting the status from the JSON
+                //The server responds with either "ok" or "error"
+                String status = jsonResponse.getString("status");
+                if (status.equals("ok")) {
+                    //If the EAN code was found, in the JSON there is an list of products, each
+                    //one with name, brand and language
+                    JSONArray productList = jsonResponse.getJSONArray("results");
+                    String[] productNames = new String[productList.length()];
+                    //Future proofing usage of brands and language, still unimplemented
+                    String[] productBrands = new String[productList.length()];
+                    String[] productLanguages = new String[productList.length()];
```

Comment:
If I were you, I've put all these declarations (from line 78 to line 81) inside the else block below. In this way you are sure there are some products in the JSON response and so all these variables instantiation is more useful.

Reply:
You are absolutely right, this will be changed soon.